# An Institution wishes to find out their student’s ability in maths, reading and writing skills. The Institution wants to do an exploratory study to check the following information.
1. Find out how many males and females participated in the test.
2. What do you think about the students' parental level of education?
3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course
4. What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course
5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.

In [44]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
data=pd.read_csv('StudentsPerformance.csv')

In [4]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [5]:
df=pd.DataFrame(data)

In [6]:
df

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# 1. Find out how many males and females participated in the test.

In [10]:
df["gender"].value_counts()

female    518
male      482
Name: gender, dtype: int64

# 2. What do you think about the students' parental level of education?

In [14]:
import statistics

In [33]:
df.groupby("parental level of education")["parental level of education"].count()

parental level of education
associate's degree    222
bachelor's degree     118
high school           196
master's degree        59
some college          226
some high school      179
Name: parental level of education, dtype: int64

In [43]:
df[['parental level of education']].describe()

parental level of education
count                         1000
unique                           6
top                   some college
freq                           226

# 3. Who scores the most on average for math, reading and writing based on 

# a)Gender

In [19]:
df1=pd.DataFrame(data,columns=['gender','math score','reading score','writing score'])

In [32]:
df1.groupby('gender').mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

# b)Test preparation course

In [26]:
df2=pd.DataFrame(data,columns=['test preparation course','math score','reading score','writing score'])

In [31]:
df2.groupby('test preparation course').mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

# 4. What do you think about the scoring variation for math, reading and writing based on

# a)Gender

In [40]:
df1[['gender','math score','reading score','writing score']].groupby('gender').var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

# b)Test preparation course

In [39]:
df2[['test preparation course','math score','reading score','writing score']].groupby('test preparation course').var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

In [42]:
df.nlargest(250,['math score'])

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]

# Inferences
* 482 Males and 518 Females participated in the test
* Regarding the "Parental level of education" ,the ones who gave the entry as 'Some college' had the most frequency(ie Mode of the dataset) and it is 226.This was followed by "Associate's degree" and "Highschool"
* For the averages for Math,Reading and writing based on gender,females fared better in reading and writing score with  averages around 72.59 and 72.47 respectively whereas the males fared better in math score with an average of around 68.82.
* For the averages for Math,Reading and writing based on test preparation course,those who had completed the test preparation course had better averages in all the three sections with averages of 69.97,74.18 and 74.68 in math,reading and writing respectively.
* Females had better scoring variance in math,reading and writing based on gender.
* Those who gave the test preparation entry as none had a better scoring variation in math,reading and writing based on test preparation.
* Among the 1000 students,25% of them(ie 250) are eligible for a bonus points based on their math score and their details have been obtained.

